File: algorithms/13.py
# 13. Roman to Integer
# https://leetcode.com/problems/roman-to-integer/

class Solution:
    def romanToInt(self, s: str) -> int:
        data = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        answer = 0
        for x, y in zip(s, s[1:]):
            if data[x] < data[y]:
                answer -= data[x]
            else:
                answer += data[x]
        return answer + data[s[-1]]
            

File: algorithms/swift/13.swift
// 13. Roman to Integer
// https://leetcode.com/problems/roman-to-integer/

class Solution {
    func romanToInt(_ s: String) -> Int {
        let data: [Character: Int] = [
            "I": 1, 
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000
        ]
        var answer: Int = 0
        var s = Array(s)
        let n: Int = s.count
        
        for i in 0..<n {
            let value: Int = data[s[i]]!
            if i + 1 < n && value < data[s[i + 1]]! {
                answer -= value
            }
            else {
                answer += value
            }
        }
        return answer
 
    }
}
